static/app/utils/useUrlParams.spec.tsx

```diff
@@ -1,82 +1,80 @@
-import type {InjectedRouter} from 'react-router';
 import {browserHistory} from 'react-router';
 import type {Location} from 'history';
 
 import {reactHooks} from 'sentry-test/reactTestingLibrary';
 
-import {RouteContext} from 'sentry/views/routeContext';
+import {useLocation} from 'sentry/utils/useLocation';
 
 import useUrlParams from './useUrlParams';
 
 jest.mock('react-router');
+jest.mock('sentry/utils/useLocation');
+
+const mockUseLocation = useLocation as jest.MockedFunction<typeof useLocation>;
+const mockBrowserHistoryPush = browserHistory.push as jest.MockedFunction<
+  typeof browserHistory.push
+>;
+
+type Query = {limit: string; page: string};
 
 describe('useUrlParams', () => {
-  type Params = {};
-  type Query = {limit: number; page: number};
-  const location = {
-    query: {
-      page: 3,
-      limit: 50,
-    },
-  } as Location<Query>;
-  const params = undefined;
-  const router = {} as InjectedRouter<Params, Query>;
-  const routes = [];
-  const wrapper = ({children}) => (
-    <RouteContext.Provider value={{location, params, router, routes}}>
-      {children}
-    </RouteContext.Provider>
-  );
+  beforeEach(() => {
+    mockBrowserHistoryPush.mockReset();
+    mockUseLocation.mockReturnValue({
+      query: {
+        page: '3',
+        limit: '50',
+      },
```

Comment:
Ah these are URL params, so they are parsed as strings

Reply:
yeah they all come back as strings. it was confusing before but i think the types are stronger in here now.

Reply:
`Location<Query>` will complain when the fields are not set to `string | string[] | undefined | null`

```
  Type 'Query' is not assignable to type 'DefaultQuery<string>'.
    Property 'limit' is incompatible with index signature.
      Type 'number' is not assignable to type 'string | string[] | null | undefined'.ts(2345)
```